draft-ietf-quic-transport.md

```diff
@@ -666,64 +664,228 @@ Version negotiation uses unprotected data.  The result of the negotiation MUST
+needed to establish a 0-RTT connection.  The idle timeout value is not expected
+to be relevant for the short time that it takes to obtain 1-RTT keys.
+
+Transport parameter values could change as a result of completing the handshake.
+The client MUST respect the new values when the handshake completes.  This
+introduces some potential problems:
+
+* A client might exceed a newly declared connection or stream flow control limit
+  with 0-RTT data.  If this occurs, the client MUST cease transmission as though
+  the flow control limit was reached.  Once WINDOW_UPDATE frames indicating an
+  increase to the affected flow control offsets is received, the client can
+  recommence sending.
+
+* Similarly, a client might exceed the concurrent stream limit declared by the
+  server.  A client MUST reset any streams that exceed this connection.  A
+  server MAY reset these streams.
+
+A client cannot use or rely upon any other transport parameter that was enabled
+in a previous connection.  This includes those defined in this document.  A
+client MUST assume that the transport parameter was absent unless the definition
+of a transport parameter provides explicit rules for use of the option in 0-RTT.
+
+
+### New Transport Parameters
+
+New transport parameters can be used to by endpoints to advertise new
+capabilities.  This can be used to negotiate new protocol behavior.  An endpoint
+MUST ignore transport parameters that it does not support.
+
+New transport parameters can be registered according to the rules in
+{{iana-transport-parameters}}.
+
+
+### Version Negotiation Validation {#version-validation}
+
+The transport parameters include three fields that encode version information.
+These retroactively authenticate the version negotiation (see
+{{version-negotiation}}) that is performed prior to the cryptographic handshake.
+
+Inclusion of these values in an extension provides integrity protection for
+these values.  As a result, modification of version negotiation packets by an
+attacker can be detected.
+
+The client includes two fields in the transport parameters that it includes in
+the TLS ClientHello:
+
+* The negotiated_version is the version that was finally selected for use.  This
+  MUST be identical to the value that is on the packet that carries the
+  ClientHello.  A server that receives a negotiated_version that does not match
+  the version of QUIC that is in use MUST terminate the connection with a TBD
+  error code.
+
+* The initial_version is the version that the client initially attempted to use.
+  If the server did not send a version negotiation packet
+  {{version-negotiation-packet}}, this will be identical to the
+  negotiated_version.
+
+A stateful server can remember how version negotiation was performed and
+validate the initial_version value.
+
+A server that does not maintain state for early packets uses a different
+process. If the initial and negotiated versions are the same, a stateless server
```

Comment:
Why are we discussing stateless servers? Version negotiation is part of the QUIC connection,  and it is not expected that the server would shoot a version negotiation packet and just drop the connection. At any rate, this is again a function change from what is, and I think belongs in a separate PR.